gpytorch/priors/lkj_prior.py

```diff
@@ -61,6 +61,58 @@ def log_prob(self, X):
         log_diag_sum = psd_safe_cholesky(X, upper=True).diagonal(dim1=-2, dim2=-1).log().sum(-1)
         return self.C + (self.eta - 1) * 2 * log_diag_sum
 
+    def _rsample(self, sample_shape=None, return_covariance=False, return_cholesky=False):
```

Comment:
The different return types here is a bit of an antipattern. Let's factor out the functionality for drawing Cholesky decomps of covariance matrices as a functional helper into sth like `gpytorch/utils/stats.py`? Then we can just call that here. (also something that would be good in a pytorch probability lib...)

Reply:
I'm not sure I follow here? The cholesky decomp of the covariance matrix is actually what's drawn in the first place, the other types end up doing various transforms of it.

Reply:
Yep indeed - I think that drawing cholesky decomps of the covariance matrix is a generally useful functionality that we should factor out rather than having hidden in this "private" `_rsample` method. Then we can just call that new method here and do the transforms if necessary.

By "antipattern" I just meant that the object `_rsample` returns depends on the input values - typically you'd consider having a different method for these but I can see how having 4 different `_rsample_*_*_` methods (e.g. `_rsample_covar_factor` being a bit much.

Reply:
We do already have four separate methods. How about refactoring the second half of that function into the various `rsample` methods by type then?

Reply:
Hmm what do you mean "We do already have four separate methods"? As part of this method? Or as part of the different Prior types? I think logically what `_rsample` does should correspond 1:1 to what the prior represents, as is the case now (e.g. if you have the `LKJCovariancePrior` then `_rsample` needs to return a sampled covar and not a factor...).

Reply:
I mean by part of the different prior types.

Reply:
Well at least the cholesky factor and the full matrix are themselves possible to reach from various prior types. I should probably code up a `LKJBaseCovariancePrior` actually to return the unscaled covariance matrices however.